gtsfm/scene_optimizer.py

```diff
@@ -171,18 +171,32 @@ def _create_process_graph(self):
 
     def _run_retriever(self, client) -> tuple[GtsfmMetricsGroup, VisibilityGraph, list[Future]]:
         retriever_start_time = time.time()
-        image_futures = self.loader.get_all_images_as_futures(client)
+        batch_size = self.image_pairs_generator._batch_size
+
+        logger.info(f"🔥 GTSFM: Loading images in batches of {batch_size}...")
+        transform = None
+
+        if self.image_pairs_generator._global_descriptor is not None:
+            transform = self.image_pairs_generator._global_descriptor.get_preprocessing_transform()
+            if transform is not None:
+                logger.info(f"🔥 GTSFM: Applying preprocessing transform for global descriptors {transform}")
+
+        # Image_Batch_Futures is a list of Stacked Tensors with dimension (batch_size, Channels, H, W)
+        image_batch_futures = self.loader.get_all_descriptor_image_batches_as_futures(client, batch_size, transform)
+
         image_fnames = self.loader.image_filenames()
 
         with performance_report(filename="dask_reports/retriever.html"):
             visibility_graph = self.image_pairs_generator.run(
                 client=client,
-                images=image_futures,
+                images=image_batch_futures,
                 image_fnames=image_fnames,
                 plots_output_dir=self.create_plot_base_path(),
             )
         retriever_metrics = self.image_pairs_generator._retriever.evaluate(len(self.loader), visibility_graph)
         retriever_duration_sec = time.time() - retriever_start_time
         retriever_metrics.add_metric(GtsfmMetric("retriever_duration_sec", retriever_duration_sec))
         logger.info("🚀 Image pair retrieval took %.2f min.", retriever_duration_sec / 60.0)
+
+        image_futures = self.loader.get_all_images_as_futures(client)
```

Comment:
I think this should go outside this function now, as it is no longer retriever related.